{agent, internal/flow}: support reasoning content discarding in agents

Rememorio opened the pull request:
- Introduced `WithReasoningContentMode` to control how `reasoning_content` is managed in multi-turn conversations, particularly for DeepSeek models.
- Added three modes: `KeepAll`, `DiscardPreviousTurns`, and `DiscardAll`, with corresponding logic in the `ContentRequestProcessor`.
- Updated tests to validate the new reasoning content modes and their effects on message processing.
- Enhanced documentation to explain the new functionality and provide usage examples for agents.

close #787

Rememorio committed:
{agent, internal/flow}: support reasoning content discarding in agents

- Introduced `WithReasoningContentMode` to control how `reasoning_content` is managed in multi-turn conversations, particularly for DeepSeek models.
- Added three modes: `KeepAll`, `DiscardPreviousTurns`, and `DiscardAll`, with corresponding logic in the `ContentRequestProcessor`.
- Updated tests to validate the new reasoning content modes and their effects on message processing.
- Enhanced documentation to explain the new functionality and provide usage examples for agents.

Rememorio committed:
linter

Rememorio committed:
comment

nomand-zc commented on internal/flow/processor/content.go:
```diff
@@ -314,6 +364,34 @@ func (p *ContentRequestProcessor) getIncrementMessages(inv *agent.Invocation, si
 
+// processReasoningContent applies reasoning content stripping based on the
+// configured mode and request boundaries.
+func (p *ContentRequestProcessor) processReasoningContent(
```
这里是为了支持interleaved thinking吗？应该有模型api的差异。 比如anthropic的sdk， 原生能支持单独传thinking字段。感觉这里是不是交给model层处理会好些？[接口](https://github.com/anthropics/anthropic-sdk-go/blob/main/message.go#L819)

Rememorio replied:
是为了支持 ds3.2 里面支持在思考阶段发生工具调用。简单来说，直到模型生成最终回复之前，应该带上 reasoning content，此为第一轮对话；模型生成最终回复之后，新开启的第二轮对话，就不要带上历史的 reasoning content 了。

仔细看了看：
1. 目前的 model 层其实有缺陷，没有带上这个字段的内容（需要 SetExtraFields 到 msg 里面）；
2. 感觉 processor 层也是有必要的，因为在 model 层里面没有 requestID，这样只能控制全部舍弃或者全部都带上，没有办法实现“舍弃前面轮次的 reasoning content”。

因此，目前比较合理的做法是：
1. 如果 reasoning content 不为空，则在 model 层带上这个 reasoning content；
2. 控制的逻辑在  processor 里面，根据策略往 model 输送 reasoning content。

Rememorio committed:
Merge branch 'main' into deepseek_3_2_reasoning_content

Rememorio committed:
feat(thinking): add debugging and tool support for reasoning content

- Introduced a new `debug` flag to print messages sent to the model API for debugging purposes.
- Implemented `reasoning_mode` flag to control handling of `reasoning_content` in multi-turn conversations.
- Added calculator and current time tools to enhance functionality in the demo.
- Updated README to reflect new features and usage instructions.
- Created a new `debug.go` file for printing chat request messages.
- Enhanced `main.go` and `tools.go` to integrate new features and improve user experience.

Rememorio committed:
test

Rememorio committed:
feat(thinking): set discard_previous_turns as default reasoning mode and update documentation

Rememorio committed:
Merge branch 'main' into deepseek_3_2_reasoning_content

WineChord reviewed (approved):
LGTM
